### ISSUE TITLE
feat: Search by plugin variant/maintainer

### ISSUE THREAD

**Issue by ReubenFrankel** [meltano/hub#1717]
it would be nice to be able to search for plugins by variant/maintainer.

![image](https://github.com/meltano/hub/assets/60552974/224b51fe-a82b-4060-858e-2661fac41731)

Also, there is an small issue where if the search term matches both a keyword and a plugin name (i.e. `matatika`), depending on the type of plugin, the one with the matching name is forced down the search results list. This feature could also address this by respecting the defined order of plugin text fields to search by (i.e. `name`, then `label`, then `variant`, and so on).

![image](https://github.com/meltano/hub/assets/60552974/e03ea52a-c407-43f1-9e54-3cf88a77c459)
![image](https://github.com/meltano/hub/assets/60552974/5e92569b-8ddc-4fc3-b270-f72d056a9036)

**Comment by edgarrmondragon** [comment]
> It would be nice to be able to search for plugins by variant/maintainer.

fwiw that's probably an easy with little changes to the graphql query and the joined text fiels in `Search.vue`

> Also, there is an small issue where if the search term matches both a keyword and a plugin name (i.e. `matatika`), depending on the type of plugin, the one with the matching name is forced down the search results list. This feature could also address this by respecting the defined order of plugin text fields to search by (i.e. `name`, then `label`, then `variant`, and so on).

but this is probably a heavier lift if I understand it correctly. Does this mean something like "if the plugin name matches the search term, then show it first regardless of plugin type"?

The problem is that for _historical_ (i.e. tech debt) reasons, search terms are  currently matched against a concatenation of a bunch of plugin fields into a single string.

**Comment by ReubenFrankel** [comment]
> fwiw that's probably an easy win with little changes to the graphql query and the joined text fiels in `Search.vue`

Yeah, I did have a look and making `variant` searchable is pretty simple, but in the case of `matatika` the actual utility with that name is sorted below a bunch of our extractors (hence the other requirement).

> Does this mean something like "if the plugin name matches the search term, then show it first regardless of plugin type"?

Yep, pretty much - maybe for `label` too. I was thinking about ordering plugins by the type of the text field (i.e. `name` or `description`) the search term best matches, but I can't say how complex that would be to implement or how performant it will be.
